addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/NetworkOceanicThingHandler.java

```diff
@@ -0,0 +1,222 @@
+                                    dataBuffer[index++] = tmpData[i];
+                                    if (logger.isTraceEnabled()) {
+                                        logger.trace("dataBuffer[{}] set to '{}'(hex '{}')", index - 1,
+                                                new String(new byte[] { dataBuffer[index - 1] }),
+                                                String.format("%02X", dataBuffer[index - 1]));
+                                    }
+                                }
+
+                                if (sequenceFound && i >= 2) {
+                                    if (tmpData[i - 2] == eChar && tmpData[i - 1] == rChar && tmpData[i] == rChar) {
+                                        // Received ERR from the device.
+                                        return null;
+                                    }
+                                }
+
+                                if (sequenceFound && i > 0
+                                        && (tmpData[i - 1] != carriageReturn && tmpData[i] == nullChar)) {
+                                    index = 0;
+                                    // Ignore trash received
+                                    if (logger.isTraceEnabled()) {
+                                        StringBuilder sb = new StringBuilder();
+                                        for (int j = 0; j < i; j++) {
+                                            sb.append(String.format("%02X", tmpData[j]));
+                                            sb.append(" ");
+                                        }
+                                        logger.trace("Ingoring {} bytes : {}", i, sb.toString());
+                                    }
+                                }
+
+                                if (sequenceFound && (tmpData[i] == carriageReturn)) {
+                                    if (index > 0) {
+                                        line = new String(Arrays.copyOf(dataBuffer, index));
+                                        logger.debug("Received response '{}'",
+                                                new String(Arrays.copyOf(dataBuffer, index)));
+                                        line = StringUtils.chomp(line);
+                                        line = line.replace(",", ".");
+                                        line = line.trim();
+                                        index = 0;
+
+                                        return line;
+                                    }
+                                }
+
+                                if (index == bufferSize) {
+                                    index = 0;
+                                }
+                            }
+                        }
+
+                    }
+                } catch (Exception e) {
+                    logger.error("An exception ocurred while quering host {}:{} : '{}'", config.ipAddress,
```

Comment:
Typo occurred check all

Reply:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
The method is also big also we can not guarantee that we are just catching communication errors because it will also catch array out bounds en NPEs